community/dbms/src/test/java/org/neo4j/commandline/dbms/MemoryRecommendationsCommandTest.java

```diff
@@ -0,0 +1,433 @@
+import static org.mockito.Matchers.argThat;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.neo4j.commandline.dbms.MemoryRecommendationsCommand.bytesToString;
+import static org.neo4j.commandline.dbms.MemoryRecommendationsCommand.recommendHeapMemory;
+import static org.neo4j.commandline.dbms.MemoryRecommendationsCommand.recommendOsMemory;
+import static org.neo4j.commandline.dbms.MemoryRecommendationsCommand.recommendPageCacheMemory;
+import static org.neo4j.configuration.ExternalSettings.initialHeapSize;
+import static org.neo4j.configuration.ExternalSettings.maxHeapSize;
+import static org.neo4j.graphdb.factory.GraphDatabaseSettings.pagecache_memory;
+import static org.neo4j.io.ByteUnit.exbiBytes;
+import static org.neo4j.io.ByteUnit.gibiBytes;
+import static org.neo4j.io.ByteUnit.mebiBytes;
+import static org.neo4j.io.ByteUnit.tebiBytes;
+
+public class MemoryRecommendationsCommandTest
+{
+    @Rule
+    public TestDirectory dir = TestDirectory.testDirectory();
+
+    @Test
+    public void mustRecommendOSMemory()
+    {
+        assertThat( recommendOsMemory( mebiBytes( 100 ) ), between( mebiBytes( 45 ), mebiBytes( 55 ) ) );
+        assertThat( recommendOsMemory( gibiBytes( 1 ) ), between( mebiBytes( 450 ), mebiBytes( 550 ) ) );
+        assertThat( recommendOsMemory( gibiBytes( 3 ) ), between( mebiBytes( 1256 ), mebiBytes( 1356 ) ) );
+        assertThat( recommendOsMemory( gibiBytes( 192 ) ), between( gibiBytes( 9 ), gibiBytes( 11 ) ) );
+        assertThat( recommendOsMemory( gibiBytes( 1920 ) ), greaterThan( gibiBytes( 29 ) ) );
+    }
+
+    @Test
+    public void mustRecommendHeapMemory()
+    {
+        assertThat( recommendHeapMemory( mebiBytes( 100 ) ), between( mebiBytes( 45 ), mebiBytes( 55 ) ) );
+        assertThat( recommendHeapMemory( gibiBytes( 1 ) ), between( mebiBytes( 450 ), mebiBytes( 550 ) ) );
+        assertThat( recommendHeapMemory( gibiBytes( 3 ) ), between( mebiBytes( 1256 ), mebiBytes( 1356 ) ) );
+        assertThat( recommendHeapMemory( gibiBytes( 6 ) ), between( mebiBytes( 3000 ), mebiBytes( 3200 ) ) );
+        assertThat( recommendHeapMemory( gibiBytes( 192 ) ), between( gibiBytes( 31 ), gibiBytes( 33 ) ) );
+        assertThat( recommendHeapMemory( gibiBytes( 1920 ) ), between( gibiBytes( 31 ), gibiBytes( 33 ) ) );
+    }
+
+    @Test
+    public void mustRecommendPageCacheMemory()
+    {
+        assertThat( recommendPageCacheMemory( mebiBytes( 100 ) ), between( mebiBytes( 95 ), mebiBytes( 105 ) ) );
+        assertThat( recommendPageCacheMemory( gibiBytes( 1 ) ), between( mebiBytes( 95 ), mebiBytes( 105 ) ) );
+        assertThat( recommendPageCacheMemory( gibiBytes( 3 ) ), between( mebiBytes( 470 ), mebiBytes( 530 ) ) );
+        assertThat( recommendPageCacheMemory( gibiBytes( 6 ) ), between( mebiBytes( 980 ), mebiBytes( 1048 ) ) );
```

Comment:
How did you come up with the particular numbers? (I ask because no 2 people seem to recommend the same thing)

Reply:
I meditated on it for a bit, and then wrote down what felt reasonable.